accounts/views.py

```diff
@@ -1,14 +1,115 @@
+class Loginview(View):
+    """docstring for Loginview"""
 
-        # if request.method == GET:
-    return render(request, "accounts/login.html")
+    def get(self, request):
+        return render(request, "accounts/login.html")
 
-    # else request.method == POST:
+    def post(self, request):
+        print("enter login post")
+        username = request.POST['username']
+        password = request.POST['password']
+        data = {
+            'username': username,
+            'password': password
+        }
+        response = requests.post(
+            'http://yashasingh.tech:8085/api/profiles/login/', data=data)
+        print(response.json())
+        data = response.json()
+        # print(data['non_field_errors'][0])
+        if data.get('non_field_errors') != None:
+            print("Incorrect credentials")
+            return render(request, "accounts/login.html")
+        if data.get('group') == "student":
+            print("student login")
+            request.session["id"] = data.get('username')
+            request.session["group"] = data.get('group')
+
+            if StudentProfile.objects.filter(
+                    username=data.get('username')).count() == 1:
+                print("already there")
+                return redirect('/dashboard')
+
+            else:
+                print("New student")
+                return redirect('/studentprofile')
+
+            return render(request, "accounts/login.html")
+        if data.get('group') == "faculty" or data.get('group') == "hod":
+            print("Faculty login")
+            request.session["id"] = data.get('username')
+            request.session["group"] = data.get('group')
+
+            if FacultyProfile.objects.filter(
+                    username=data.get('username')).count() == 1:
+                print("already there")
+                return redirect('/facultydashboard')
+
+            else:
+                print('New faculty')
+                print("Faculty")
+                return redirect('/facultyprofile')
+            # print("Faculty login")
+        # if data.get('group') == "hod":
+        #     request.session["id"] = data.get('username')
+        #     request.session["group"] = data.get('group')
+
+        #     # render (account/studentprofile.html)
+        #     print("HOD login")
+
+
+class Studentprofileupdate(View):
+    # @student_only
+
+    def get(self, request):
+        return render(request, "accounts/studentprofile.html")
+
+    def post(self, request):
+        print("enter student post")
+        # username = request.POST.get('username')
+        branch = request.POST.get('branch')
+        email = request.POST.get('email')
+        Phone = request.POST.get('phone')
+        StudentProfile.objects.create(
+            username=request.session['id'],
+            group=branch, Branch=branch,
+            email=email, phone=Phone)
+        return HttpResponse("succesful")
+
+
+class Facultyprofileupdate(View):
```

Comment:
If anyone hits the URL for this view, ANYONE CAN CREATE A FACULTY PROFILE!
Provide the required authentication check here!!